yoti-sdk-api/src/main/java/com/yoti/api/client/ReceiptDetails.java

```diff
@@ -0,0 +1,121 @@
+package com.yoti.api.client;
+
+import java.util.Date;
+import java.util.Optional;
+
+public final class ReceiptDetails {
+
+    private final String id;
+    private final String sessionId;
+    private final String rememberMeId;
+    private final String parentRememberMeId;
+    private final ApplicationProfile profile;
+    private final HumanProfile otherPartyProfile;
+    private final Date date;
+    private final ExtraData extraData;
+
+    private ReceiptDetails(Builder builder) {
+        id = builder.id;
+        sessionId = builder.sessionId;
+        rememberMeId = builder.rememberMeId;
+        parentRememberMeId = builder.parentRememberMeId;
+        profile = builder.profile;
+        otherPartyProfile = builder.otherPartyProfile;
+        date = builder.date;
+        extraData = builder.extraData;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getSessionId() {
+        return sessionId;
+    }
+
+    public String getRememberMeId() {
+        return rememberMeId;
+    }
+
+    public String getParentRememberMeId() {
+        return parentRememberMeId;
+    }
+
+    public ApplicationProfile getProfile() {
+        return profile;
+    }
+
+    public HumanProfile getOtherPartyProfile() {
+        return otherPartyProfile;
+    }
```

Comment:
All of these values are "optional", i.e. a receipt _might_ not contain them.

This would be better expressed if the return values were wrapped in `Optional`

Although from a technical perspective, given the way the `ReceiptDetails` instance is built the "profiles" won't be `null` they can be "empty" (as in they contain no data).

One option would be to only set them if they contain data (so if "empty" the value would be left `null`.